packages/mock-addon/src/utils/faker.js

```diff
@@ -119,10 +119,20 @@ export class Faker {
 
         if (matched) {
             const { response, status, delay = 0 } = matched;
+
             return new Promise((resolve) => {
                 setTimeout(() => {
                     if (typeof response === 'function') {
-                        resolve(new Response(url, status, response(request)));
+                        const data = response(request);
+
+                        let status = 200;
+                        const customStatusCode = data?.header?.status;
+                        if (customStatusCode) {
+                            status = customStatusCode;
+                        }
+                        delete data.header;
```

Comment:
shouldn't we pass the header?

Reply:
This is to allow the status to be customizable inside the function. 
And at the same time leaving the rest of the code alone. The status might be important to mock, the rest of the header not so.
This allows to set the status code based on conditionals in the response function. The current status in the main header is not passed in the request, so you cannot determine how to set anything at that level